lib/query.js

```diff
@@ -1618,6 +1630,11 @@ Query.prototype.setOptions = function(options, overwrite) {
     this._mongooseOptions.defaults = options.defaults;
     // deleting options.defaults will cause 7287 to fail
   }
+  if ('translateAliases' in options) {
+    this._mongooseOptions.translateAliases = options.translateAliases;
+    delete options.translateAliases;
```

Comment:
Does this mutate the provided options object?

Reply:
No, there's a shallow clone on L1590